gensim/models/tfword2vec.py

```diff
@@ -0,0 +1,210 @@
+        self.window_size = window_size
+
+        # The minimum number of word occurrences for it to be included in the
+        # vocabulary.
+        self.min_count = min_count
+
+        # Subsampling threshold for word occurrence.
+        self.subsample = subsample
+
+        # Where to write out summaries.
+        self.save_path = save_path
+
+        # Eval options.
+
+        # The text file for eval.
+        self.eval_data = eval_data
+
+
+def modified_tfw2v_init(self, options, session):
+    self.options = options
+    self.session = session
+    self.word2id = {}
+    self.id2word = []
+    self.build_graph()
+    self.build_eval_graph()
+    if options.save_path is not None:
+        self.save_vocab()
+    if options.eval_data is not None:
+        self.read_analogies()
+
+#TODO take care of monkey patching.
+Word2Vec.__init__ = modified_tfw2v_init
+
+
+class TfWord2Vec(GensimWord2VecNoTraining):
+
+    def __init__(self, train_data=None, save_path=None, eval_data=None,
+                 embedding_size=200, epochs_to_train=15, learning_rate=0.025,
+                 num_neg_samples=25, batch_size=500, concurrent_steps=12,
+                 window_size=5, min_count=5, subsample=1e-3):
+
+        self.options = Options(train_data, save_path=save_path, eval_data=eval_data,
+                               embedding_size=embedding_size, epochs_to_train=epochs_to_train,
+                               learning_rate=learning_rate, num_neg_samples=num_neg_samples,
+                               batch_size=batch_size, concurrent_steps=concurrent_steps,
+                               window_size=window_size, min_count=min_count, subsample=subsample)
+
+        self.convert_input(train_data)
+        self.train()
+        self.vocab = {}
+        self.create_vocab()
+
+    def train(self):
+        with tf.Graph().as_default(), tf.Session() as session:
+            self.model = Word2Vec(self.options, session)
+            for _ in xrange(self.options.epochs_to_train):
+                self.model.train()  # Process one epoch
+                if self.options.eval_data is not None:
+                    self.model.eval()  # Eval analogies.'''
+
+            self.syn0 = self.model._w_in
+            self.syn0norm = session.run(tf.nn.l2_normalize(self.model._w_in, 1))
+            self.index2word = self.model._id2word
+
+    def create_vocab(self):
+        for word in self.options.vocab_words:
+            self.vocab[word] = Vocab(index=self.model._word2id[word])
+
+    def convert_input(self, corpus):
+        """
+        Converts gensim corpus to a file that can be used by tf word2vec
+
+        """
+        #assumes that the string represents a file extension
+        if not isinstance(corpus, string_types):
+            fout = tempfile.TemporaryFile()
+            for line in corpus:
+                for word in line:
+                    fout.write(utils.to_utf8(str(word) + " "))
```

Comment:
Calling `str` on unicode strings is not a good idea (will encode all non-ASCII character with blackslashes... probably not what you wanted).

Also, join the strings with `u' '.join()`, rather than this.